src/generators/static/tsConfigFileGenerator.ts

```diff
@@ -4,32 +4,56 @@
 import { Project } from "ts-morph";
 import { getAutorestOptions } from "../../autorestSession";
 
+const highLevelTsConfig = {
+  compilerOptions: {
+    module: "es6",
+    moduleResolution: "node",
+    strict: true,
+    target: "es5",
+    sourceMap: true,
+    declarationMap: true,
+    esModuleInterop: true,
+    allowSyntheticDefaultImports: true,
+    forceConsistentCasingInFileNames: true,
+    preserveConstEnums: true,
+    lib: ["es6", "dom"],
+    declaration: true,
+    outDir: "./esm",
+    importHelpers: true
+  },
+  include: ["./src/**/*.ts"],
+  exclude: ["node_modules"]
+};
+
+const restLevelTsConfig = {
+  compilerOptions: {
+    module: "commonjs",
+    target: "es2015",
+    declaration: true,
+    declarationMap: true,
+    outDir: "./esm"
+  },
+  exclude: [
+    "node_modules",
+    "types",
+    "temp",
+    "browser",
+    "dist",
+    "dist-esm",
+    "./samples/**/*.ts"
+  ]
+};
+
 export function generateTsConfig(project: Project) {
-  const { generateMetadata } = getAutorestOptions();
+  const { generateMetadata, restLevelClient } = getAutorestOptions();
```

Comment:
NIT: I am wonder if we should generalize `restLevelClient` to be more like an enum instead of a boolean

Reply:
What values do you think we should get? Something like `--client-type=<"rest" | "highlevel">`?

Reply:
I think that would be great

Reply:
It looks like all other languages are using the boolean flag. It is also less prone to typos. I think I'll keep it this way and if we need to generate other kind of clients in the future we can turn it to an enum like

Reply:
Fair!